aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/BaseHandlerStd.java

```diff
@@ -399,6 +399,39 @@ protected boolean isDBClusterMember(final ResourceModel model) {
         return StringUtils.isNotBlank(model.getDBClusterIdentifier());
     }
 
+    protected ResourceModel restoreIdentifier(final ResourceModel observed, final ResourceModel original) {
```

Comment:
Why we return ResourceModel here, Should we make this method `void` as it is editing value in observed object only and always returning observed?

Reply:
I decided to stick to the functional interface as the new CFN framework is somewhat promoting it all around. In this case it simplifies passing the arguments around as in ReadHandler:
```java
return ProgressEvent.success(restoreIdentifier(Translator.translateDbInstanceFromSdk(dbInstance), request.getDesiredResourceState()), context);
```
You are right: this is not a real functional approach as it comes with a side effect. Given that all use cases of these methods converge to passing a brand new object that was transformed from the response, I took this shortcut. In other words, I'm not concerned about having this side effect atm, but if I had to be 100% pedantic about it, I'd stick to a functional approach with a modified copy of the object returned back. Hope it makes sense.